externals/llvm-external-projects/torch-mlir-dialects/lib/Conversion/TcpToLinalg/Elementwise.cpp

```diff
@@ -174,24 +174,108 @@ createLinalgPayloadForElementwiseOp(Operation *op,
     else if (elemType.isa<mlir::IntegerType>())
       return {b.create<arith::MulIOp>(loc, payloadArgs[0], payloadArgs[1])};
     else
-      llvm_unreachable(
-          "unsupported element type in createLinalgPayloadForElementwiseOp");
+      llvm_unreachable("unsupported element type in "
+                       "createLinalgPayloadForElementwiseOp for tcp.mul");
   }
 
   if (isa<DivFOp>(op)) {
     if (elemType.isa<mlir::FloatType>())
       return {b.create<arith::DivFOp>(loc, payloadArgs[0], payloadArgs[1])};
     else
-      llvm_unreachable(
-          "unsupported element type in createLinalgPayloadForElementwiseOp");
+      llvm_unreachable("unsupported element type in "
+                       "createLinalgPayloadForElementwiseOp for tcp.divf");
   }
 
   if (isa<Atan2Op>(op)) {
     if (elemType.isa<mlir::FloatType>())
       return {b.create<math::Atan2Op>(loc, payloadArgs[0], payloadArgs[1])};
     else
-      llvm_unreachable(
-          "unsupported element type in createLinalgPayloadForElementwiseOp");
+      llvm_unreachable("unsupported element type in "
+                       "createLinalgPayloadForElementwiseOp for tcp.atan2");
+  }
+
+  if (auto castOp = dyn_cast<CastOp>(op)) {
+    auto inputType =
+        castOp.getIn().getType().dyn_cast<RankedTensorType>().getElementType();
+    auto outputType = resultTensorType.getElementType();
+    // check for same type
+    if (inputType.getIntOrFloatBitWidth() ==
+            outputType.getIntOrFloatBitWidth() &&
+        ((!castOp.getInDtype() && !castOp.getOutDtype()) ||
+         (castOp.getInDtype() && castOp.getOutDtype() &&
+          castOp.getInDtype().value() == castOp.getOutDtype().value())))
```

Comment:
nit: since you are adding `{ }` for the else cases below, it is better to add that here as well, to keep it uniform and more readable.